python/cudf/cudf/io/parquet.py

```diff
@@ -16,6 +16,26 @@
 from cudf.utils import ioutils
 from cudf.utils.utils import _cudf_nvtx_annotate
 
+BYTE_SIZES = {
+    "kB": 10**3,
+    "MB": 10**6,
+    "GB": 10**9,
+    "TB": 10**12,
+    "PB": 10**15,
+    "KiB": 2**10,
+    "MiB": 2**20,
+    "GiB": 2**30,
+    "TiB": 2**40,
+    "PiB": 2**50,
+    "B": 1,
+    "": 1,
+}
+BYTE_SIZES = {k.lower(): v for k, v in BYTE_SIZES.items()}
+BYTE_SIZES.update(
+    {k[0]: v for k, v in BYTE_SIZES.items() if k and "i" not in k}
+)
+BYTE_SIZES.update({k[:-1]: v for k, v in BYTE_SIZES.items() if k and "i" in k})
```

Comment:
If we're going to start with a hardcoded dict why not just hardcode the result of this processing?

```
{'': 1,
 'b': 1,
 'g': 1000000000,
 'gb': 1000000000,
 'gi': 1073741824,
 'gib': 1073741824,
 'k': 1000,
 'kb': 1000,
 'ki': 1024,
 'kib': 1024,
 'm': 1000000,
 'mb': 1000000,
 'mi': 1048576,
 'mib': 1048576,
 'p': 1000000000000000,
 'pb': 1000000000000000,
 'pi': 1125899906842624,
 'pib': 1125899906842624,
 't': 1000000000000,
 'tb': 1000000000000,
 'ti': 1099511627776,
 'tib': 1099511627776}
```

Reply:
Changed this to a pre-populated dict.